connect/runtime/src/main/java/org/apache/kafka/connect/util/ConnectUtils.java

```diff
@@ -65,4 +68,12 @@ static String lookupKafkaClusterId(Admin adminClient) {
                                        + "Check worker's broker connection and security properties.", e);
         }
     }
+
+    public static void addMetricsContextProperties(Map<String, Object> prop, WorkerConfig config, String clusterId) {
+        //add all properties predefined with "metrics.context."
+        prop.putAll(config.originalsWithPrefix(CommonClientConfigs.METRICS_CONTEXT_PREFIX, false));
+        //add connect properties
+        prop.put(CommonClientConfigs.METRICS_CONTEXT_PREFIX + WorkerConfig.CONNECT_KAFKA_CLUSTER_ID, clusterId);
+        prop.put(CommonClientConfigs.METRICS_CONTEXT_PREFIX + WorkerConfig.CONNECT_GROUP_ID, config.originals().get(DistributedConfig.GROUP_ID_CONFIG));
```

Comment:
Standalone does not have a `group.id` property, since each standalone worker is its own single-node cluster. That means that this line can result in a null value for the `metrics.context.connect.group.id` entry in the `props` map.

Unfortunately, there is no identifier for standalone workers. Is the `metrics.context.connect.group.id` property really required if the Standalone worker doesn't use any kind of coordination? It seems from [other changes in `ConnectMetrics`](https://github.com/apache/kafka/pull/8691/files#diff-bb4c06b85c34f39a0c6bb771c4566759R90-R92) that it is not required, in which case we should update the KIP to reflect that this property is only added in Connect distributed mode and we should correct the logic above to only add this property if it is non-null similar to the [changes in `ConnectMetrics`](https://github.com/apache/kafka/pull/8691/files#diff-bb4c06b85c34f39a0c6bb771c4566759R90-R92).

If it is required, then we'll have to figure out something else.

Reply:
Speaking of this, it'd probably be worth adding unit tests for this method with both `StandaloneConfig` and `DistributedConfig` objects to verify this method adds the right properties to the supplied map.

Reply:
Added unit test in ConnectUtilsTest for StandAloneConfig and DistributedConfig.

Reply:
I'll leave KIP update to @xvrl

Reply:
@rhauch I've updated the KIP document, will send a follow-up email to announce the changes, once we finalize @mumrah 's feedback